src/systemTest/java/com/hivemq/cli/utils/cli/MqttCliAsyncExtension.java

```diff
@@ -43,9 +43,10 @@ public class MqttCliAsyncExtension implements AfterEachCallback {
     private @Nullable Path temporaryHomeDir = null;
 
     @Override
-    public void afterEach(final @NotNull ExtensionContext context) throws IOException {
+    public void afterEach(final @NotNull ExtensionContext context) throws IOException, InterruptedException {
         for (final Process startedProcess : startedProcesses) {
             startedProcess.destroyForcibly();
+            startedProcess.waitFor();
```

Comment:
We have to ensure the cli process is terminated, otherwise we will have a race condition with the [list of connect packets ](https://github.com/hivemq/mqtt-cli/blob/6bf7cdd2005cbf7dcc96e5743edbeb652a0d0857/src/systemTest/java/com/hivemq/cli/utils/broker/HiveMQExtension.java#L126)that we keep track of. The process might still make connect calls before it is fully terminated. 

The other alternative is stop using the `test` command for these tests altogether as we only care about the connect command for these tests. We abruptly terminate the process after performing the assertions. 

For now, I have gone with the simplest change that avoids the race condition.

Reply:
Nice idea to do it here. Does this affect the test duration negatively?

Reply:
There was no visible performance degradation.